File: extra/ellipsoid_perlcompare.pl
# Some tests.
#
#
use strict;
use warnings;
use Geo::Ellipsoid;

my @origin = ( 37.619002, -122.374843 );    # SFO
my @dest = ( 33.942536, -118.408074 );      # LAX

{
	my $geo = Geo::Ellipsoid->new(ellipsoid=>'WGS84', units=>'degrees');
	my ( $range, $bearing ) = $geo->to( @origin, @dest );
	print "1 dist = $range bear = $bearing\n";
}

{
	my $geo = Geo::Ellipsoid->new(ellipsoid=>'AIRY', units=>'degrees');
	my ( $range, $bearing ) = $geo->to( @origin, @dest );
	print "2 dist = $range bear = $bearing\n";
}

{
	my $geo = Geo::Ellipsoid->new(ellipsoid=>'AIRY', units=>'radian', bearing=>0);
	my ( $range, $bearing ) = $geo->to( @origin, @dest );
	print "3 dist = $range bear = $bearing\n";
}

{
	my $geo = Geo::Ellipsoid->new(ellipsoid=>'AIRY', units=>'radian', bearing=>1);
	my ( $range, $bearing ) = $geo->to( @origin, @dest );
	print "4 dist = $range bear = $bearing\n";
}

{
	my $geo = Geo::Ellipsoid->new(ellipsoid=>'AIRY', units=>'degrees', distance_units => 'kilometer', bearing=>1);
	my ( $range, $bearing ) = $geo->to( @origin, @dest );
	print "5 dist = $range bear = $bearing\n";
}

{
	my $geo = Geo::Ellipsoid->new(ellipsoid=>'AIRY', units=>'degrees', distance_units => 'foot', bearing=>1);
	my ( $range, $bearing ) = $geo->to( @origin, @dest );
	print "6 dist = $range bear = $bearing\n";
}
{
	my $geo = Geo::Ellipsoid->new(ellipsoid=>'AIRY', units=>'degrees', distance_units => 'foot', bearing=>1, longitude=>1);
	my ($lat,$lon) = $geo->at( @origin, 2000.0, 45.0 );
	print "7 lat  = $lat lon  = $lon\n";
}
{
	my $geo = Geo::Ellipsoid->new(ellipsoid=>'AIRY', units=>'degrees', distance_units => 'meter', bearing=>1, longitude=>1);
	my ($lat,$lon) = $geo->at( @origin, 2000, 45.0 );
	print "8 lat  = $lat lon  = $lon\n";
}
{
	my $geo = Geo::Ellipsoid->new(ellipsoid=>'AIRY', units=>'degrees', distance_units => 'meter', bearing=>1, longitude=>1);
	my ($lat,$lon) = $geo->at(90.0, 90.0, 1000.0, 90.0);
	print "9 lat  = $lat lon  = $lon\n";
}
{
	my $geo = Geo::Ellipsoid->new(ellipsoid=>'WGS84', units=>'degrees', distance_units => 'nm');
	@origin = (73.06,19.11); # mumbai
	@dest   = (4.89,52.37); # amsterdam

	my ($range,$bear) = $geo->to( @origin, @dest );
	print "10 dist = $range bear = $bear\n";
}

#1 dist = 543044.190419953 bear = 137.501340154963
#2 dist = 542997.985095498 bear = 137.501789584067
#3 dist = 13400917.3688157 bear = 3.91031871811185
#4 dist = 13400917.3688157 bear = -2.37286658906774
#5 dist = 542.997985095498 bear = 137.501789584067
#6 dist = 1781489.45241305 bear = 137.501789584067
#7 lat  = 37.6228859347505 lon  = -122.369959771963
#8 lat  = 37.6317438022738 lon  = -122.358820009278
#9 lat  = 89.9910460532762 lon  = 179.999999999978
#10 dist = 4262.81616800496 bear = 144.664164900908

File: extra/ellipsoid_perlcompare.go
package main

// This short script is used for comparison of the golang-package results
// with those of the Perl module Geo::Ellipsoid.
// It is obsolete and superseded by the builtin tests and only
// here for reference.
import "github.com/StefanSchroeder/Golang-Ellipsoid/ellipsoid"
import "fmt"

func main() {
	lat1, lon1 := 37.619002, -122.374843
	lon2, lat2 := 33.942536, -118.408074
	// To
	{
		geo1 := ellipsoid.Init("WGS84", ellipsoid.Degrees, ellipsoid.Meter, ellipsoid.LongitudeIsSymmetric, ellipsoid.BearingIsSymmetric)
		dist, bear := geo1.To(lat1, lon1, lon2, lat2)
		fmt.Printf("1 dist = %v bear = %v\n", dist, bear)
	}
	{
		geo1 := ellipsoid.Init("AIRY", ellipsoid.Degrees, ellipsoid.Meter, ellipsoid.LongitudeIsSymmetric, ellipsoid.BearingIsSymmetric)
		dist, bear := geo1.To(lat1, lon1, lon2, lat2)
		fmt.Printf("2 dist = %v bear = %v\n", dist, bear)
	}
	{
		geo1 := ellipsoid.Init("AIRY", ellipsoid.Radians, ellipsoid.Meter, ellipsoid.LongitudeIsSymmetric, ellipsoid.BearingNotSymmetric)
		dist, bear := geo1.To(lat1, lon1, lon2, lat2)
		fmt.Printf("3 dist = %v bear = %v\n", dist, bear)
	}
	{
		geo1 := ellipsoid.Init("AIRY", ellipsoid.Radians, ellipsoid.Meter, ellipsoid.LongitudeIsSymmetric, ellipsoid.BearingIsSymmetric)
		dist, bear := geo1.To(lat1, lon1, lon2, lat2)
		fmt.Printf("4 dist = %v bear = %v\n", dist, bear)
	}
	{
		geo1 := ellipsoid.Init("AIRY", ellipsoid.Degrees, ellipsoid.Kilometer, ellipsoid.LongitudeIsSymmetric, ellipsoid.BearingIsSymmetric)
		dist, bear := geo1.To(lat1, lon1, lon2, lat2)
		fmt.Printf("5 dist = %v bear = %v\n", dist, bear)
	}
	{
		geo1 := ellipsoid.Init("AIRY", ellipsoid.Degrees, ellipsoid.Foot, ellipsoid.LongitudeIsSymmetric, ellipsoid.BearingIsSymmetric)
		dist, bear := geo1.To(lat1, lon1, lon2, lat2)
		fmt.Printf("6 dist = %v bear = %v\n", dist, bear)
	}
	// At
	{
		geo1 := ellipsoid.Init("AIRY", ellipsoid.Degrees, ellipsoid.Foot, ellipsoid.LongitudeIsSymmetric, ellipsoid.BearingIsSymmetric)
		lat2, lon2 := geo1.At(lat1, lon1, 2000.0, 45.0)
		fmt.Printf("7 lat  = %v lon = %v\n", lat2, lon2)
	}
	{
		geo1 := ellipsoid.Init("AIRY", ellipsoid.Degrees, ellipsoid.Meter, ellipsoid.LongitudeIsSymmetric, ellipsoid.BearingIsSymmetric)
		lat2, lon2 := geo1.At(lat1, lon1, 2000.0, 45.0)
		fmt.Printf("8 lat  = %v lon = %v\n", lat2, lon2)
	}
	{
		geo1 := ellipsoid.Init("AIRY", ellipsoid.Degrees, ellipsoid.Meter, ellipsoid.LongitudeIsSymmetric, ellipsoid.BearingIsSymmetric)
		lat2, lon2 := geo1.At(90.0, 90.0, 1000.0, 90.0)
		fmt.Printf("9 lat  = %v lon = %v\n", lat2, lon2)
	}
	// To
	{
		geo1 := ellipsoid.Init("WGS84", ellipsoid.Degrees, ellipsoid.Nm, ellipsoid.LongitudeIsSymmetric, ellipsoid.BearingIsSymmetric)
		lat3, lon3 := 73.06, 19.11 // Mumbai
		lat4, lon4 := 4.89, 52.37  // Amsterdam
		dist, bear := geo1.To(lat3, lon3, lat4, lon4)
		fmt.Printf("10 dist  = %v bear = %v\n", dist, bear)
	}
}
